cedar-policy-core/src/ast/policy.rs

```diff
@@ -761,10 +764,10 @@ impl StaticPolicy {
             resource_constraint,
             non_head_constraints,
         );
-        let num_slots = body.condition().slots().next().map(SlotId::clone);
+        let first_slot = body.condition().slots().next();
         // INVARIANT (inline policy correctness), checks that no slots exists
-        match num_slots {
-            Some(slot_id) => Err(UnexpectedSlotError::FoundSlot(slot_id))?,
+        match first_slot {
+            Some(slot) => Err(UnexpectedSlotError::FoundSlot(slot.id))?,
```

Comment:
I don't think this error is `pub use`'ed, so could we add the loc to it also?

Reply:
this simple idea led to a larger refactor.  Re-review needed.